pennylane/transforms/metric_tensor.py

```diff
@@ -400,27 +425,38 @@ def _metric_tensor_cov_matrix(tape, diag_approx):
     obs_list = []
     coeffs_list = []
     params_list = []
+    in_argnum_list = []
 
     for queue, curr_ops, param_idx, _ in graph.iterate_parametrized_layers():
         params_list.append(param_idx)
+        in_argnum_list.append([p in argnum for p in param_idx])
+
+        if not any(in_argnum_list[-1]):
+            # no tape needs to be created for this block
+            continue
+
         coeffs_list.append([])
         obs_list.append([])

```

Comment:
I don't understand this. What do you mean by "working with the inner lists"?

Reply:
Ah, I meant that we could just create the list `this_coeffs = []` and `this_obs = []`, append the content to them (instead of appending to `obs_list[-1]` and `coeffs_list[-1]`, which indexes the outer lists `obs_list` and `coeffs_list` every time), and at the end append `this_coeffs` and `this_obs` to `coeffs_list` and `obs_list`. 

Sorry, not sure how clearly I am expressing myself here. :laughing:

Reply:
Ah, I understand. I agree. I just adopted the way it was done before. I changed it now. Can you see whether you think it's good now.